### PR TITLE
fixed serious bug in CAP_INTERNAL_GET_CORRESPONDING_OUTPUT_OBJECTS

### PR DESCRIPTION
**Pull request by mohamed-barakat**
causing

ProjectionInFactorOfFiberProductWithGivenFiberProduct( [ mor1, mor2 ], 1, P );

to return

ProjectionInFactorOfFiberProductWithGivenFiberProduct( [ mor1, mor2 ], 2, P );

### CODE REVIEW

File: CAP/gap/ToolsForCategories.gi [hunk 1]
```diff
@@ -686,7 +686,7 @@ InstallGlobalFunction( CAP_INTERNAL_GET_CORRESPONDING_OUTPUT_OBJECTS,
             if ForAll( current_output[ 2 ], i -> i in "0123456789" ) then
                 list_position := Int( current_output[ 2 ] );
             else
-                list_position := Position( input_list, current_output[ 2 ] );
+                list_position := function_input[ Position( input_list, current_output[ 2 ] ) ];
```
Comment (sebastianpos) [inline CAP/gap/ToolsForCategories.gi]: If `Position` returns `fail` then we don't have a meaningful error message anymore.
I suggest to write
```
list_position := function_input[ list_position ];
```
right after the `list_position = fail` statement, i.e., in line 695.

Comment (mohamed-barakat) [reply CAP/gap/ToolsForCategories.gi]: There are two assignments of `list_position` in the `if-else` statement. The first one is correct and the second was buggy.

Your suggestion will also alter the first correct assignment and break it.

Comment (sebastianpos) [reply CAP/gap/ToolsForCategories.gi]: Yes, you are right. But this is easy to overcome. Could you please fix it?

Comment (mohamed-barakat) [reply CAP/gap/ToolsForCategories.gi]: I added a commit with an error message.

**Pull request merged**